iosHyperskillApp/iosHyperskillApp/Sources/Modules/TrackSelectionList/Views/TrackSelectionListView.swift

```diff
@@ -0,0 +1,172 @@
+        .onAppear {
+            viewModel.startListening()
+            viewModel.onViewAction = handleViewAction(_:)
+        }
+        .onDisappear {
+            viewModel.stopListening()
+            viewModel.onViewAction = nil
+        }
+    }
+
+    // MARK: Private API
+
+    @ViewBuilder
+    private func buildBody() -> some View {
+        switch viewModel.viewStateKs {
+        case .idle:
+            TrackSelectionListSkeletonView()
+                .onAppear(perform: viewModel.doLoadTrackSelectionList)
+        case .loading:
+            TrackSelectionListSkeletonView()
+        case .error:
+            PlaceholderView(
+                configuration: .networkError(
+                    backgroundColor: appearance.backgroundColor,
+                    action: viewModel.doRetryLoadTrackSelectionList
+                )
+            )
+        case .content(let viewData):
+            ScrollView {
+                VStack(spacing: appearance.spacing) {
+                    TrackSelectionListHeaderView()
+
+                    TrackSelectionListGridView(
+                        tracks: viewData.tracks,
+                        onTrackTap: viewModel.doMainTrackAction(trackID:)
+                    )
+                }
+                .padding()
+            }
+            .frame(maxWidth: .infinity)
+        }
+    }
+}
+
+// MARK: - TrackSelectionListView (ViewAction) -
+
+private extension TrackSelectionListView {
+    func handleViewAction(_ viewAction: TrackSelectionListFeatureActionViewAction) {
+        switch TrackSelectionListFeatureActionViewActionKs(viewAction) {
+        case .navigateTo(let navigateToViewAction):
+            handleNavigateToViewAction(
+                navigateToViewAction
+            )
+        case .showTrackSelectionConfirmationModal(let showTrackSelectionConfirmationModalViewAction):
+            handleShowTrackSelectionConfirmationModalViewAction(
+                showTrackSelectionConfirmationModalViewAction
+            )
+        case .showTrackSelectionStatus(let showTrackSelectionStatusViewAction):
+            handleShowTrackSelectionStatusViewAction(
+                showTrackSelectionStatusViewAction
+            )
+        }
+    }
+
+    func handleNavigateToViewAction(_ viewAction: TrackSelectionListFeatureActionViewActionNavigateTo) {
+        switch TrackSelectionListFeatureActionViewActionNavigateToKs(viewAction) {
+        case .studyPlan:
+            TabBarRouter(
+                tab: .studyPlan,
+                popToRoot: true
+            )
+            .route()
+        }
+    }
+
+    func handleShowTrackSelectionConfirmationModalViewAction(
+        _ viewAction: TrackSelectionListFeatureActionViewActionShowTrackSelectionConfirmationModal
+    ) {
+        guard let rootViewController = stackRouter.rootViewController else {
+            return
+        }
+
+        let trackID = viewAction.track.id
+
+        let alertController = UIAlertController(
```

Comment:
Is it useful to create child of UIAlertController and add some delegation protocol to move this code like we do it for pan modals?

Reply:
@vladkash Yes, it would be nice to have, I think in the future we can create a builder for alerts, for example see [this PR](https://github.com/GitHawkApp/GitHawk/pull/431).